nixos/modules/config/update-users-groups.pl

```diff
@@ -235,7 +235,11 @@ sub parseUser {
     # Ensure home directory incl. ownership and permissions.
     if ($u->{createHome} and !$is_dry) {
         make_path($u->{home}, { mode => oct($u->{homeMode}) }) if ! -e $u->{home};
-        chown $u->{uid}, $u->{gid}, $u->{home};
+        if ($u->{recursiveCreateHome}) {
+            chown -r $u->{uid}, $u->{gid}, $u->{home};
```

Comment:
I don't think this is valid perl code. Here the documentation for the chown function: https://perldoc.perl.org/functions/chown

Reply:
I think you would have to use `system("chown", "-R", ...)`: https://perldoc.perl.org/functions/system

Reply:
yeah It seems like that's the case (god I fucking hate perl I knew touching it was a mistake) 
do I have to do anything special to get the gnu core utils binary into the path that perl system reads from?
I'm also a bit buffeld by the fact that the nixos test didn't catch it because I would have expected the perl thing to just fail when recursiveCreateHome is set to true but what do I know and maybe the script has some error handling.

Reply:
I think the activation script already relies heavily on coreutils.

Reply:
Should be fine now.